Build script is broken

git commit id
commit 3b5054bb6a48a657552164f27ac6953f6895a55a (HEAD -> main, origin/main, origin/HEAD)


$ sudo ./lib/build_lib.sh
CMake Error: The source directory "/home/thkrishn/poseidonos" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
Makefile:261: /home/thkrishn/poseidonos/lib/spdk/mk/spdk.deps.mk: No such file or directory
make: *** No rule to make target '/home/t.krishna/poseidonos/lib/spdk/mk/spdk.deps.mk'.  Stop.

hello even After I go in to the lib directory I see these errors
patch fails (why not use spdk as a git submodule?)

```
CMake Error at /home/thkrishn/poseidonos/lib/fio-prefix/src/fio-stamp/fio-patch-.cmake:37 (message):
  Command failed: 2

   'patch' '-t' '-p0' '<' '/home/thkrishn/poseidonos/lib/fio-3.1.patch'

  See also

    /home/thkrishn/poseidonos/lib/fio-prefix/src/fio-stamp/fio-patch.log


-- Log output is:
patching file '<'
Hunk #1 FAILED at 121.
1 out of 1 hunk FAILED -- saving rejects to file '<.rej'
patch: **** Can't reopen file '<' : No such file or directory

CMake Error at /home/thkrishn/poseidonos/lib/fio-prefix/src/fio-stamp/fio-patch-.cmake:47 (message):
  Stopping after outputting logs.


make[2]: *** [CMakeFiles/fio.dir/build.make:100: fio-prefix/src/fio-stamp/fio-patch] Error 1
make[1]: *** [CMakeFiles/Makefile2:347: CMakeFiles/fio.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
-- spdlog download command succeeded.  See also /home/thkrishn/poseidonos/lib/spdlog-prefix/src/spdlog-stamp/spdlog-download-*.log
[ 17%] Performing update step for 'spdlog'
[ 18%] Performing patch step for 'spdlog'
CMake Error at /home/thkrishn/poseidonos/lib/spdlog-prefix/src/spdlog-stamp/spdlog-patch-.cmake:37 (message):
  Command failed: 2

   'patch' '-t' '-p0' '<' '/home/thkrishn/poseidonos/lib/spdlog-1.4.2.patch'

  See also

    /home/thkrishn/poseidonos/lib/spdlog-prefix/src/spdlog-stamp/spdlog-patch.log


-- Log output is:
patching file '<'
Hunk #1 FAILED at 124.
Hunk #2 FAILED at 140.
Hunk #3 FAILED at 152.
Hunk #4 FAILED at 160.
4 out of 4 hunks FAILED -- saving rejects to file '<.rej'
patching file '<'
Hunk #1 FAILED at 12.
1 out of 1 hunk FAILED -- saving rejects to file '<.rej'
patching file '<'
Hunk #1 FAILED at 11.
Hunk #2 FAILED at 19.
2 out of 2 hunks FAILED -- saving rejects to file '<.rej'
patching file '<'
Hunk #1 FAILED at 889.
Hunk #2 FAILED at 1216.
2 out of 2 hunks FAILED -- saving rejects to file '<.rej'
patching file '<'
Hunk #1 FAILED at 104.
Hunk #2 FAILED at 170.
2 out of 2 hunks FAILED -- saving rejects to file '<.rej'
patching file '<'
Hunk #1 FAILED at 287.
Hunk #2 FAILED at 311.
2 out of 2 hunks FAILED -- saving rejects to file '<.rej'
patching file '<'
Hunk #1 FAILED at 37.
1 out of 1 hunk FAILED -- saving rejects to file '<.rej'
patch: **** Can't reopen file '<' : No such file or directory

CMake Error at /home/thkrishn/poseidonos/lib/spdlog-prefix/src/spdlog-stamp/spdlog-patch-.cmake:47 (message):
  Stopping after outputting logs.


make[2]: *** [CMakeFiles/spdlog.dir/build.make:100: spdlog-prefix/src/spdlog-stamp/spdlog-patch] Error 1
make[2]: *** Waiting for unfinished jobs....
-- spdlog update command succeeded.  See also /home/thkrishn/poseidonos/lib/spdlog-prefix/src/spdlog-stamp/spdlog-update-*.log
make[1]: *** [CMakeFiles/Makefile2:374: CMakeFiles/spdlog.dir/all] Error 2
-- dpdk download command succeeded.  See also /home/thkrishn/poseidonos/lib/dpdk-prefix/src/dpdk-stamp/dpdk-download-*.log
[ 20%] Performing update step for 'dpdk'
[ 21%] Performing patch step for 'dpdk'
CMake Error at /home/thkrishn/poseidonos/lib/dpdk-prefix/src/dpdk-stamp/dpdk-patch-.cmake:37 (message):
  Command failed: 2

   'patch' '-t' '-p0' '<' '/home/thkrishn/poseidonos/lib/dpdk-20.08.patch'

  See also

    /home/thkrishn/poseidonos/lib/dpdk-prefix/src/dpdk-stamp/dpdk-patch.log
```

--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -103,7 +103,7 @@ ExternalProject_Add(dpdk
     LOG_INSTALL ON
     LOG_OUTPUT_ON_FAILURE ON
     LOG_MERGED_STDOUTERR ON
-    PATCH_COMMAND patch -t -p0 < ${PROJ_ROOT}/lib/dpdk-20.08.patch
+    PATCH_COMMAND patch -t -p0 -i ${PROJ_ROOT}/lib/dpdk-20.08.patch
     CONFIGURE_COMMAND "" 
     BUILD_COMMAND make install MAKE_PAUSE=n T=${TARGET} EXTRA_CFLAGS=${EXTRA_CFLAGS} DESTDIR=. -j 4
     INSTALL_COMMAND "")
@@ -131,7 +131,7 @@ ExternalProject_Add(fio
     LOG_INSTALL ON
     LOG_OUTPUT_ON_FAILURE ON
     LOG_MERGED_STDOUTERR ON
-    PATCH_COMMAND patch -t -p0 < ${PROJ_ROOT}/lib/fio-3.1.patch
+    PATCH_COMMAND patch -t -p0 -i ${PROJ_ROOT}/lib/fio-3.1.patch
     CONFIGURE_COMMAND "" 
     BUILD_COMMAND make -j ${NUM_BUILD_CORE})
 
@@ -167,7 +167,7 @@ ExternalProject_Add(spdk
     LOG_INSTALL ON
     LOG_OUTPUT_ON_FAILURE ON
     LOG_MERGED_STDOUTERR ON
-    PATCH_COMMAND patch -t -p0 < ${PROJ_ROOT}/lib/spdk-20.10.patch
+    PATCH_COMMAND patch -t -p0 -i ${PROJ_ROOT}/lib/spdk-20.10.patch
     CONFIGURE_COMMAND ./configure --with-dpdk=${PROJ_ROOT}/lib/${POS_DEP_DPDK} --with-rdma --with-fio=${PROJ_ROOT}/lib/${POS_DEP_FIO} --with-pos --without-isal --without-vhost --with-pmdk ${ENABLE_DEBUG}
     BUILD_COMMAND ./scripts/pkgdep.sh
     COMMAND make -j ${NUM_BUILD_CORE}
@@ -259,7 +259,7 @@ ExternalProject_Add(spdlog
     LOG_INSTALL ON
     LOG_OUTPUT_ON_FAILURE ON
     LOG_MERGED_STDOUTERR ON
-    PATCH_COMMAND patch -t -p0 < ${PROJ_ROOT}/lib/spdlog-1.4.2.patch
+    PATCH_COMMAND patch -t -p0 -i ${PROJ_ROOT}/lib/spdlog-1.4.2.patch
     CONFIGURE_COMMAND cmake -H. -B_builds -DCMAKE_INSTALL_PREFIX=. -DCMAKE_BUILD_TYPE=Release -DCMAKE_POSITION_INDEPENDENT_CODE=ON
     BUILD_COMMAND cmake --build _builds --target install
     INSTALL_COMMAND "")

--- a/lib/build_grpc.sh
+++ b/lib/build_grpc.sh
@@ -101,6 +101,15 @@ main()
 	rm -rf grpc
 	git clone ${GRPC_REPO}
 	cd grpc
+	if [ ${BUILD_INTERNAL} == FALSE ]
+	then
+		echo "Cleaning up existing dirs in grpc/third_party/(re2, abseil-cpp, protobuf)"
+		rm -rf third_party/re2 third_party/abseil-cpp third_party/protobuf
+
+		echo "Checking out GRPC submodules..."
+		git submodule update --init
+	fi
+
 	install_re2
 	install_protobuf
 	install_abseil

--- a/src/telemetry/telemetry_air/telemetry_air_delegator.cpp
+++ b/src/telemetry/telemetry_air/telemetry_air_delegator.cpp
@@ -60,6 +60,7 @@ TelemetryAirDelegator::StartDelegation(void)
         returnState = stateRun;
     }
 
+#ifdef TELEMETRY_FOR_AIR_IS_ENABLED
     air_request_data({"PERF_ARR_VOL"},
         [this](const air::JSONdoc& data) -> int {
             const std::lock_guard<std::mutex> lock(this->mutex);
@@ -98,6 +99,7 @@ TelemetryAirDelegator::StartDelegation(void)
             }
             return this->returnState;
         });
+#endif
 }
 
 void

